mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java

```diff
@@ -62,6 +66,37 @@
 @Contract(MapperProvider.class)
 public class MessageMapperTest<T extends MapperProvider> {
 
+    private class ConcurrentSetFlagTestRunnable implements Runnable {
+        private final int threadNumber;
+        private final int updateCount;
+        private final Mailbox mailbox;
+        private final MessageUid uid;
+        private final CountDownLatch countDownLatch;
+
+        public ConcurrentSetFlagTestRunnable(int threadNumber, int updateCount, Mailbox mailbox, MessageUid uid, CountDownLatch countDownLatch) {
+            this.threadNumber = threadNumber;
+            this.updateCount = updateCount;
+            this.mailbox = mailbox;
+            this.uid = uid;
+            this.countDownLatch = countDownLatch;
+        }
+
+        @Override
+        public void run() {
+            countDownLatch.countDown();
+            for (int i = 0; i < updateCount; i++) {
+                try {
+                    messageMapper.updateFlags(mailbox,
+                        new FlagsUpdateCalculator(new Flags("custom-" + threadNumber + "-" + i),
+                            FlagsUpdateMode.ADD),
+                        MessageRange.one(uid));
```

Comment:
Maybe you can extract all but this call in a lamba which could be a parameter to ConcurrentSetFlagTestRunnable and then make this one shared between the two test classes.